service-commands/scripts/A

```diff
@@ -119,218 +31,585 @@ def create_instance(name, azure, image_project, image_family, size, machine_type
+        sys.exit(1)
 
-    if service in ["creator-node", "discovery-provider"]:
-        print("Setting up audius-k8s-manifests...")
+    return proc.stdout.decode().strip()
+
+
+def _setup_audius_k8s_manifests(provider, user, config, name):
+    """Setup audius-k8s-manifests on a remote machine"""
+    ssh_args = _get_ssh_args(provider, name, user)
+
+    success = (
         subprocess.run(
             [
-                "ssh",
-                host,
+                *ssh_args,
                 (
-                    "git clone https://github.com/AudiusProject/audius-k8s-manifests.git && "
-                    "cd audius-k8s-manifests && "
-                    "yes | sh setup.sh "
+                    "git clone https://github.com/AudiusProject/audius-k8s-manifests.git &&"
+                    "yes | sh audius-k8s-manifests/setup.sh"
                 ),
-            ]
-        )
+            ],
+            check=False,
+        ).returncode
+        == 0
+    )
+    if not success:
+        return False
 
-        print("Waiting for instance to restart")
-        returncode = None
-        while returncode != 0:
-            returncode = subprocess.run(["ssh", host, ":"])
+    print("Waiting for instance to restart")
+    _wait_for_instance(provider, name, user)
+
+    if not config:
+        print("Warning no audius-k8s-config specified", file=sys.stderr)
+        return True
 
-        if config:
-            print("Copying config...")
+    with open(config, encoding="locale") as config_file:
+        success = (
             subprocess.run(
-                ["ssh", host, "cat > $MANIFESTS_PATH/config.yaml"],
-                input=open(config).read(),
-            )
+                [*ssh_args, "cat > $MANIFESTS_PATH/config.yaml"],
+                input=config_file.read(),
+                check=False,
+            ).returncode
+            == 0
+        )
+
+    return success
+
+
+def _configure_local_hosts(ip_addr):
+    """Configure local /etc/hosts file"""
+    return (
+        subprocess.run(
+            [
+                "sudo",
+                "-E",
+                f"AUDIUS_REMOTE_DEV_HOST={ip_addr}",
+                "node",
+                f"{os.getenv('PROTOCOL_DIR', '.')}/service-commands/scripts/hosts.js",
+                "add-remote-host",
+            ],
+            check=False,
+        ).returncode
+        == 0
+    )
+
+
+def _create_azure_instance(image, disk_size, machine_type, name):
+    """Create an instance on azure"""
+    image = image or DEFAULT_AZURE_IMAGE
+    machine_type = machine_type or DEFAULT_AZURE_MACHINE_TYPE
+    disk_size = str(disk_size)
+
+    print("\033[1mProvider:       \033[0m", "Azure")
+    print("\033[1mAccount:        \033[0m", _get_azure_account())
+    print("\033[1mSubscription:   \033[0m", _get_azure_subscription())
+    print("\033[1mResource Group: \033[0m", _get_azure_resource_group())
+    print("\033[1mOS Image:       \033[0m", image)
```

Comment:
instead of having magical chars like this in the code can we define a `pretty_print` python fn and hide this detail in there?